packages/react-components/react-combobox/src/components/Option/useOption.tsx

```diff
@@ -31,16 +31,17 @@ function getValueString(value: string | undefined, children: React.ReactNode) {
  * @param ref - reference to root HTMLElement of Option
  */
 export const useOption_unstable = (props: OptionProps, ref: React.Ref<HTMLElement>): OptionState => {
-  const { disabled, value } = props;
+  const { data, disabled, value } = props;
   const optionRef = React.useRef<HTMLElement>(null);
   const optionValue = getValueString(value, props.children);
 
   // use the id if provided, otherwise use a generated id
   const id = useId('fluent-option', props.id);
 
   // data used for context registration & events
-  const optionData = React.useMemo<OptionValue>(() => ({ id, disabled, value: optionValue }), [
+  const optionData = React.useMemo<OptionValue>(() => ({ id, data, disabled, value: optionValue }), [
```

Comment:
you also use `optionData` in an effect, if users don't memoize their `data` it risks running the effect more than you might expect